helper.cpp

```diff
@@ -247,7 +285,8 @@ const char* exiv2_iptc_datum_key(const Exiv2IptcDatum *datum)
 
 const char* exiv2_iptc_datum_to_string(const Exiv2IptcDatum *datum)
 {
-	return datum->datum.toString(0).c_str();
+	const std::string strval = datum->datum.toString();
+	return strdup(strval.c_str());
```

Comment:
this is added, because otherwise non-ascii strings were corrupted when reading them (see an example of a non-ascii string in test files)